flytekit/clients/auth/authenticator.py

```diff
@@ -293,7 +296,36 @@ def __init__(
             verify=verify,
         )
 
+    def _initialize_auth_client(self):
+        if not self._auth_client:
+            self._set_header_key(self._header_key)
+            self._auth_client = AuthorizationClient(
+                endpoint=self._endpoint,
+                redirect_uri=self._redirect_uri,
+                client_id=self._client_id,
+                audience=self._audience,
+                scopes=self._scopes,
+                auth_endpoint=self._authorization_endpoint,
+                token_endpoint=self._token_endpoint,
+                verify=self._verify,
+                session=self._session,
+                refresh_access_token_params={},
+            )
+
     def refresh_credentials(self):
+        self._initialize_auth_client()
+        if self._creds:
+            """We have an access token so lets try to refresh it"""
+            try:
+                self._creds = self._auth_client.refresh_access_token(self._creds)
```

Comment:
Based on the spec you linked that seems like it should be ok? I validated this on our setup and it worked fine for us and we use `offline_access`.

> Typically this will not be included in the request, and if omitted, the service should issue an access token with the same scope as was previously issued.

Reply:
Ah okay, non-issue then. I do recall encountering issues where not including this scope caused problems, but I wonder if that has to do with Azure AD, which could be different than Okta.